File: 3408-count-the-number-of-special-characters-i/count-the-number-of-special-characters-i.py
class Solution:
    def numberOfSpecialChars(self, word: str) -> int:
        lowercaseCount = [False] * 26
        uppercaseCount = [False] * 26
        for c in word:
            if "a" <= c <= "z":
                lowercaseCount[ord(c) - ord("a")] = True
            elif "A" <= c <= "Z":
                uppercaseCount[ord(c) - ord("A")] = True
        ans = 0
        for i in range(26):
            if lowercaseCount[i] and uppercaseCount[i]:
                ans += 1

        return ans

File: 3408-count-the-number-of-special-characters-i/count-the-number-of-special-characters-i.cpp
class Solution {
public:
    int numberOfSpecialChars(string word) {
        bool lower[26] = {};
        bool upper[26] = {};

        for (char& c : word) {
            if ('a' <= c && c <= 'z') {
                lower[c - 'a'] = true;
            } else {
                upper[c - 'A'] = true;
            }
        }

        int ans = 0;
        for (int i = 0; i < 26; ++i) {
            if (lower[i] && upper[i]) {
                ++ans;
            }
        }
        return ans;
    }
};